### ISSUE TITLE
M1 macOS Monterey 12.5, SDL2 Controller Mapping causes Crash

### ISSUE THREAD

**Issue by HomeStarRunnerTron** [RPCS3/rpcs3#14883]
### Quick summary

I have an 8BitDo Pro 2 controller that I began using with RPCS3 recently, and I noticed that, at the same time I started using it, every single time I tried to hit Stop on a game I was playing, it'd crash with this kind of error attached:

```
Illegal instruction at 00007ff80191761e (0x0f0b90488d0531e82000488905e1c355).
Thread: Main Thread.
```

or...

```
Segfault reading location 00003eaddd4d9018 at 00007ff8015ac71d.
Thread: Main Thread.
```

Something along those lines.

I assumed it was from a recent RPCS3 update, so I tried to downgrade a bit, only to experience the same issue. Finally, I stopped using my controller and just booted the game with Keyboard controls again, hit Stop, and-- no crash happened! Consistently, every time I stopped playing a game, it'd be fine. I decided to try starting a game using my 8BitDo Pro 2 controller through SDL2, then right before I was going to stop the game, I switched the Pad configuration to Keyboard. No crash, again!

From what I can tell, it seems to dislike anything that utilizes SDL2. If I try to add in some undetected controller like a DolphinBar or a MayFlash PS controller by adding it into gamecontrollerdb.txt, it will barely ever open up the Pads settings without force-quitting right there. Don't know if this is just because I'm on the macOS version I'm on, but just thought I'd bring it up in-case anyone else is experiencing similar issues and wants to know why.

### Details

Currently running 0.0.29-15568-105c6949, but I've tried the most recent Mac build (0.0.29-15786), and no difference in this issue.

### Attach a log file

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/13626856/RPCS3.log)

### System configuration

macOS Monterey
Version 12.5
MacBook Pro (14-inch, 2021)
Apple M1 Pro
16 GB Memory

**Comment by erivaldodantas** [comment]
Is the same for me

**Comment by Airetex** [comment]
Having the same issue over here as well. I'm trying to connect a switch pro controller

**Comment by mavethee** [comment]
This can be closed, after Qt 6.6.2 PR being merged it's no longer an issue.
